### PR TITLE
fix(duplicate id validations):

### PR DESCRIPTION
**Pull request by daniel7764**
-Now you can't move a contact to complete without checking that id has no duplicates

### CODE REVIEW

File: db scripts/check_duplicates_ids.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
+CREATE OR REPLACE FUNCTION public.check_duplicates_ids(curridentificationnumber character varying, investigationid integer, interactedcontactid integer)
+ RETURNS boolean
+ LANGUAGE plpgsql
+AS $function$
+declare
+	foundDuplicates boolean = false;
+	begin
+			if exists(select 1
```
Comment (RoeiRom) [inline db scripts/check_duplicates_ids.sql:8]: Why not using select into?

**Review by RoeiRom — CHANGES_REQUESTED** [review]

File: db scripts/check_duplicates_ids.sql (hunk 1 above)
Comment (daniel7764) [reply db scripts/check_duplicates_ids.sql:8]: i don't know, that's how daniel landau does it

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo.tsx [hunk 2]
```diff
@@ -46,20 +46,18 @@ const ContactQuestioningInfo: React.FC<Props> = (props: Props): JSX.Element => {
...
-                        saveContact({...interactedContact, contactStatus: selectedStatus?.id});
-                    } else {
-                        alertWarning(`שים לב, מספר זיהוי ${interactedContact.identificationNumber} חוזר על עצמו, אנא בצע את השינויים הנדרשים`);
-                        setPreviousContactStatus();
-                    }
+                    axios.post('/contactedPeople/checkDuplicates', {
+                        currIdNumber: interactedContact.identificationNumber,
+                        personId: interactedContact.id
+                    })
+                        .then((result) => {
```
Comment (MayaSasson) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo.tsx:53]: I wouldn’t do the duplicate check in the first place if the is is empty

**Review by RoeiRom — APPROVED** [review]

**Review by nevobonneh — APPROVED** [review]

**Review by roymaniv — APPROVED** [review]

**Review by nevobonneh — APPROVED** [review]

File: client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo.tsx [hunk 3]
```diff
@@ -46,19 +46,21 @@ const ContactQuestioningInfo: React.FC<Props> = (props: Props): JSX.Element => {
                 confirmButtonText: 'כן, המשך'
             }).then((result) => {
                 if (result.value) {
-                    let foundDuplicateIds: boolean = false;
                     if (interactedContact.identificationNumber !== '') {
-                        axios.post('/contactedPeople/checkDuplicates', {currIdNumber: interactedContact.identificationNumber})
+                        axios.post('/contactedPeople/checkDuplicates', {
+                            currIdNumber: interactedContact.identificationNumber,
+                            personId: interactedContact.id
```
Comment (sandymash) [inline client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo.tsx:52]: maybe use a more readable name..
the difference between currIdNumber and personId isnt very clear?

Comment (daniel7764) [reply client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactQuestioningInfo.tsx]: fixed

**Review by nevobonneh — APPROVED** [review]

**Review by sandymash — APPROVED** [review]

**Pull request merged**